refinery/tool_manager/models.py

```diff
@@ -449,40 +456,127 @@ class Meta:
 
-    def create_collection_description(self):
+    @property
+    def galaxy_collection_type(self, nesting_string=""):
         """
-        Creates a dict containing information describing the Galaxy DataSet
-        Collection to create
-        :return: dict
+        Get the string representation of the Galaxy Dataset Collection type
+        from a WorkflowTool's nested file_relationships.
+        :param nesting_string: Placeholder string for recursive iterations to
+        append to.
+        :return: <String> in the following form: ^((list|paired):*)+[^:]+$
         """
-        nesting = self._get_nesting_string()
-        return {
-            'name': 'Collection ({})'.format(nesting),
-            "collection_type": nesting,
-            "element_identifiers": [
-                {
-                    'id': file_mapping[self.GALAXY_DATASET_HISTORY_ID],
-                    'name': "Refinery FileStoreItem: {}".format(
-                        file_mapping[self.REFINERY_FILE_UUID]
-                    ),
-                    'src': 'hda'
-                }
-                for file_mapping in self.get_file_relationships_galaxy()
-            ]
-        }
+        flattened_nesting = self._flatten_file_relationships_nesting()
+        for index, file_relationship in enumerate(flattened_nesting):
+            if isinstance(file_relationship, list):
+                nesting_string += "{}:".format(WorkflowTool.LIST)
+            if isinstance(file_relationship, tuple):
+                nesting_string += "{}:".format(WorkflowTool.PAIRED)
+        return nesting_string[:-1]
+
+    @property
+    def galaxy_history_id(self):
+        return self.get_galaxy_dict()[self.GALAXY_IMPORT_HISTORY_DICT]["id"]
+
+    def _associate_collection_elements(self, galaxy_element_data):
+        """
+        Handles the association of Galaxy objects with their parent elements
+        :param galaxy_element_data: list of bioblend objects to associate
+        with each other
+        :return: <CollectionDescription>
+        """
+        if not len(galaxy_element_data) == 1:
+            for index, galaxy_element in enumerate(galaxy_element_data):
+                if type(galaxy_element) == CollectionDescription:
+                    for prev_galaxy_element in galaxy_element_data[:index]:
+                        if (type(prev_galaxy_element) ==
+                                HistoryDatasetElement or
+                                type(prev_galaxy_element) ==
+                                CollectionElement):
+                            galaxy_element.add(prev_galaxy_element)
+                            galaxy_element_data.remove(prev_galaxy_element)
+                            return self._associate_collection_elements(
+                                galaxy_element_data
+                            )
+                if type(galaxy_element) == CollectionElement:
+                    for prev_galaxy_element in galaxy_element_data[:index]:
```

Comment:
There should be logic here to account for `forward` and `reverse` `HistoryDatasetElement`s when the `CollectionType` is `paired`